code/game/dna/genes/vg_powers.dm

```diff
@@ -39,69 +39,99 @@
+		head_organ.g_facial = hex2num(copytext(new_facial, 4, 6))
+		head_organ.b_facial = hex2num(copytext(new_facial, 6, 8))
+
+	//Head accessory.
+	if(head_organ.species.bodyflags & HAS_HEAD_ACCESSORY)
+		var/list/valid_head_accessories = M.generate_valid_head_accessories()
+		var/new_head_accessory = input("Please select head accessory style", "Character Generation", head_organ.ha_style) as null|anything in valid_head_accessories
+		if(new_head_accessory)
+			head_organ.ha_style = new_head_accessory
+
+		var/new_head_accessory_colour = input("Please select head accessory colour.", "Character Generation", rgb(head_organ.r_headacc, head_organ.g_headacc, head_organ.b_headacc)) as null|color
+		if(new_head_accessory_colour)
+			head_organ.r_headacc = hex2num(copytext(new_head_accessory_colour, 2, 4))
+			head_organ.g_headacc = hex2num(copytext(new_head_accessory_colour, 4, 6))
+			head_organ.b_headacc = hex2num(copytext(new_head_accessory_colour, 6, 8))
+
+	//Body markings.
+	if(M.species.bodyflags & HAS_MARKINGS)
+		var/list/valid_markings = M.generate_valid_markings()
+		var/new_marking = input("Please select marking style", "Character Generation", M.m_style) as null|anything in valid_markings
+		if(new_marking)
+			M.m_style = new_marking
+
+		var/new_marking_colour = input("Please select marking colour.", "Character Generation", rgb(M.r_markings, M.g_markings, M.b_markings)) as null|color
+		if(new_marking_colour)
+			M.r_markings = hex2num(copytext(new_marking_colour, 2, 4))
+			M.g_markings = hex2num(copytext(new_marking_colour, 4, 6))
+			M.b_markings = hex2num(copytext(new_marking_colour, 6, 8))
+
+	//Body accessory.
+	if(M.species.tail && M.species.bodyflags & HAS_TAIL)
+		var/list/valid_body_accessories = M.generate_valid_body_accessories()
+		if(valid_body_accessories.len > 1) //By default valid_body_accessories will always have at the very least a 'none' entry populating the list, even if the user's species is not present in any of the list items.
+			var/new_body_accessory = input("Please select body accessory style", "Character Generation", M.body_accessory) as null|anything in valid_body_accessories
+			if(new_body_accessory)
+				M.body_accessory = body_accessory_by_name[new_body_accessory]
+
+	//Skin tone.
+	if(M.species.bodyflags & HAS_SKIN_TONE)
+		var/new_tone = input("Please select skin tone level: 1-220 (1=albino, 35=caucasian, 150=black, 220='very' black)", "Character Generation", "[M.s_tone]") as null|text
+		if (!new_tone)
+			new_tone = 35
+		M.s_tone = 35 - max(min(round(text2num(new_tone)), 220), 1)
+
+	//Skin colour.
+	if(M.species.bodyflags & HAS_SKIN_COLOR)
+		var/new_body_colour = input("Please select body colour.", "Character Generation", rgb(M.r_skin, M.g_skin, M.b_skin)) as null|color
+		if(new_body_colour)
+			M.r_skin = hex2num(copytext(new_body_colour, 2, 4))
+			M.g_skin = hex2num(copytext(new_body_colour, 4, 6))
+			M.b_skin = hex2num(copytext(new_body_colour, 6, 8))
+
+	M.force_update_limbs()
```

Comment:
Part of the fix for the bug where skin colour wouldn't change when you used morph.